src/main/kotlin/org/stellar/walletsdk/util/StellarToml.kt

```diff
@@ -0,0 +1,123 @@
+package org.stellar.walletsdk.util
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import org.stellar.sdk.Server
+import org.stellar.walletsdk.NetworkRequestFailedException
+import org.stellar.walletsdk.StellarTomlAddressMissingHomeDomain
+import org.stellar.walletsdk.StellarTomlMissingFields
+import shadow.com.moandjiezana.toml.Toml
+
+/**
+ * [Stellar info file](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0001.md)
+ * (also known as TOML file) methods.
+ *
+ * @property stellarAddress Stellar address whose TOML file to fetch
+ * @property server Horizon [Server] instance
+ * @property httpClient optional custom HTTP client, uses [OkHttpClient] by default
+ */
+class StellarToml(
+  private val stellarAddress: String,
```

Comment:
Why does the `StellarToml` class require a Stellar account as a parameter? Most anchors don't issue their own asset, so they don't have issuer accounts, and most wallets won't know the distribution accounts of the anchors they integrate with.

I think a more natural parameter would be `homeDomain`.

Reply:
Thanks for catching this! I oversimplified it to make it easier on the UI (getting everything from the asset). I'll update to require `homeDomain` here.